src/main/flight/pid.c

```diff
@@ -287,23 +293,60 @@ void pidResetIterm(void)
     }
 }
 
+
 #ifdef USE_WING
-static float getWingTpaArgument(float throttle)
+static float calcWingThrottle(void)
+{
+    float batteryThrottleFactor = 1.0f;
+    if (pidRuntime.tpaSpeed.maxVoltage > 0.0f) {
+        batteryThrottleFactor = getBatteryVoltageLatest() / 100.0f / pidRuntime.tpaSpeed.maxVoltage;
+        batteryThrottleFactor = constrainf(batteryThrottleFactor, 0.0f, 1.0f);
+    }
+
+    return getRootMeanSquareMotorOutput() * batteryThrottleFactor;
+}
+
+
+static float calcWingAcceleration(float throttle, float pitchAngleRadians)
 {
-    const float pitchFactorAdjustment = scaleRangef(throttle, 0.0f, 1.0f, pidRuntime.tpaGravityThr0, pidRuntime.tpaGravityThr100);
-    const float pitchAngleFactor = getSinPitchAngle() * pitchFactorAdjustment;
-    DEBUG_SET(DEBUG_TPA, 1, lrintf(pitchAngleFactor * 1000.0f));
+    const tpaSpeedParams_t *tpa = &pidRuntime.tpaSpeed;
 
-    float tpaArgument = throttle + pitchAngleFactor;
-    const float maxTpaArgument = MAX(1.0 + pidRuntime.tpaGravityThr100, pidRuntime.tpaGravityThr0);
-    tpaArgument = tpaArgument / maxTpaArgument;
-    tpaArgument = pt2FilterApply(&pidRuntime.tpaLpf, tpaArgument);
-    DEBUG_SET(DEBUG_TPA, 2, lrintf(tpaArgument * 1000.0f));
+    const float thrust = (throttle * throttle - throttle * tpa->speed * tpa->inversePropMaxSpeed) * tpa->twr * G_ACCELERATION;
+    const float drag = tpa->speed * tpa->speed / tpa->massDragRatio;
```

Comment:
(`*  tpa->inverseMassDragRatio`)

Reply:
i guess it would be then `dragMassRatio`.
why would it be better here instead of division?

Reply:
Division takes up to 14 CPU cycles. Multiplication only takes 4

Reply:
It was more about allowing zero drag, similar to pitch speed. But there is slight improvement in performance too